pkg/v20/resource/appmigration/create.go

```diff
@@ -62,3 +70,52 @@ func (r *Resource) newChartSpecsToMigrate() []key.ChartSpec {
 
 	return chartSpecsToMigrate
 }
+
+func addCordonAnnotations() map[string]string {
+	return map[string]string{
+		annotation.CordonReason:    "cordoning chartconfig CR for migration to app CR",
+		annotation.CordonUntilDate: time.Now().Add(1 * time.Hour).Format("2006-01-02T15:04:05"),
+	}
+}
+
+func addDeleteAnnotation() map[string]string {
+	return map[string]string{
+		annotation.DeleteCustomResourceOnly: "true",
+	}
+}
+
+func patchChartConfig(tenantG8sClient versioned.Interface, chartCR v1alpha1.ChartConfig, annotations map[string]string) error {
+	patches := []Patch{}
+
+	if len(chartCR.Annotations) == 0 {
+		patches = append(patches, Patch{
+			Op:    "add",
+			Path:  "/metadata/annotations",
+			Value: map[string]string{},
+		})
+	}
+
+	for k, v := range annotations {
+		patches = append(patches, Patch{
+			Op:    "add",
+			Path:  fmt.Sprintf("/metadata/annotations/%s", replaceToEscape(k)),
+			Value: v,
+		})
+	}
+
+	bytes, err := json.Marshal(patches)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	_, err = tenantG8sClient.CoreV1alpha1().ChartConfigs("giantswarm").Patch(chartCR.Name, types.JSONPatchType, bytes)
```

Comment:
Not big but we are patching here without checking the `resourceVersion` - not really a problem here as we aren't concurrently altering chartconfigs or their annotations _yet_ :)

Reply:
I thought checking the resource version was for update rather than patch API calls? This is using JSON patch so the annotations are added but otherwise the CR is not modified.

Reply:
Since this is only `patch` operation we don't need to check whole resourceVersion for it.